### PR TITLE
Add result to presentPaywall methods

### PR DESCRIPTION
**Pull request by tonidero**
Adds a `PaywallResult` enum and makes it the return value of the `presentPaywall` and `presentPaywallIfNeeded` methods

### CODE REVIEW

File: revenuecat_examples/purchase_tester/lib/src/app.dart [hunk 1]
```diff
@@ -147,7 +148,8 @@ class _UpsellScreenState extends State<UpsellScreen> {
 
         buttonThings.add(ElevatedButton(
           onPressed: () async {
-            await RevenueCatUI.presentPaywall();
+            final paywallResult = await RevenueCatUI.presentPaywall();
+            log('Paywall result: $paywallResult');
```
Comment (tonidero) [inline revenuecat_examples/purchase_tester/lib/src/app.dart:152]: Right now I'm only logging this. I did run into some issues getting this log to show for iOS, so I used the `dart:developer` log method for that which fixed the issue for me.

**Comment by tonidero** [comment]
Still need to update PHC with the fixes in https://github.com/RevenueCat/purchases-hybrid-common/pull/639 but tested locally and seems to be working fine now!

File: purchases_ui_flutter/lib/paywall_result.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/// Possible values for the result of a paywall presentation.
+enum PaywallResult {
+  /// Only returned when using [RevenueCatUI.presentPaywallIfNeeded]. Returned
+  /// if the paywall was not presented.
+  notPresented,
+  /// Returned when the paywall was presented and the user cancelled without
+  /// executing an action.
+  cancelled,
```
Comment (NachoSoto) [inline purchases_ui_flutter/lib/paywall_result.dart:8]: What if you tap on purchase, cancel, and then purchase again? Would you only get this?

Comment (tonidero) [reply purchases_ui_flutter/lib/paywall_result.dart:8]: You would only get the last result before the paywall closed. In this case, the purchase.

Comment (NachoSoto) [reply purchases_ui_flutter/lib/paywall_result.dart:8]: Ooooh when it's closed, got it 👍🏻

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**